xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -22,10 +22,13 @@
 
 #include "PlatformDefs.h"
 
-#define COMSKIP_HEADER "FILE PROCESSING COMPLETE"
-#define VIDEOREDO_HEADER "<Version>2"
-#define VIDEOREDO_TAG_CUT "<Cut>"
-#define VIDEOREDO_TAG_SCENE "<SceneMarker "
+namespace
+{
+constexpr const char* ComskipHeader = "FILE PROCESSING COMPLETE";
```

Comment:
```suggestion
constexpr std::string_view ComskipHeader = "FILE PROCESSING COMPLETE";
```
if you make this `std::string_view` you don't need to use `strncmp` and can use a c++ compare instead. Same thing goes for the other constexpr defines

Reply:
I was not aware I could use `std::string_view` already, nice.
Unfortunately this is far more complicated than I'd antecipate. Not all strncp calls are direct comparisons, most of the time we just want to know if the constexpr is contained within the buffer. Did you mean `std::compare`?
If so I am stuck here:

```
[build] /home/arch/Github/xbmc/xbmc/cores/VideoPlayer/Edl.cpp:402:21: error: request for member ‘compare’ in ‘buffer.std::array<char, 1024>::data()’, which is of non-class type ‘std::array<char, 1024>::pointer’ {aka ‘char*’}
[build]   402 |       buffer.data().compare(0, VideoredoHeader.size(), VideoredoHeader) != 0)
[build]       |                     ^~~~~~~
```

Reply:
Help with this is greatly appreciated as I don't know how to proceed here. Otherwise I suggest to keep this with `const char*` for now, I sort of depend on this PR to continue working on other stuff. Not that it includes any functionality but I'll have to rebase this multiple times. Would like to get this one in before touching anything in the EDL file.